### DISCUSSION TITLE
Can I invoke promises with `@xstate/fsm`?

### DISCUSSION THREAD

**Discussion by frederikhors** [statelyai/xstate#2521] [Q&A] [answered]
I'm having a lot of issues using `@xstate/fsm`.

I don't understand if I can https://xstate.js.org/docs/guides/communication.html#invoking-promises with it.

Can I?

**Comment by davidkpiano** [comment, marked as answer]
You cannot. Please see here to see a comparison: https://xstate.js.org/docs/packages/xstate-fsm/

**Reply by frederikhors** [reply]
Yes I saw it. But because I'm not an expert I cannot understand from there.

I only want a way to call a promise with resolve and reject states. Should I use the big xstate? Really?

**Reply by davidkpiano** [reply]
For now, yes. We're exploring ways to have simple invocations for `@xstate/fsm` though.
